Signal/src/ViewControllers/ProfileViewController.m

```diff
@@ -29,28 +44,47 @@ @interface ProfileViewController () <UITextFieldDelegate, AvatarViewHelperDelega
 
 @property (nonatomic) BOOL hasUnsavedChanges;
 
+@property (nonatomic) ProfileViewMode profileViewMode;
+
 @end
 
 #pragma mark -
 
 @implementation ProfileViewController
 
+- (instancetype)initWithMode:(ProfileViewMode)profileViewMode
+{
+    self = [super init];
+
+    if (!self) {
+        return self;
+    }
+
+    self.profileViewMode = profileViewMode;
+
+    // Use the TSStorageManager.dbReadWriteConnection for consistency with the reads below.
+    [[[TSStorageManager sharedManager] dbReadWriteConnection] setDate:[NSDate new]
```

Comment:
I agree that we should be moving away from the global db connections, hence the way I had it before.  But I don't want to open a new db connection _that we never close_ everywhere we do a write.  Why don't we talk at standup.